packages/dd-trace/src/llmobs/sdk.js

```diff
@@ -0,0 +1,448 @@
+        }
+
+        if (result && kind !== 'retrieval' && !LLMObsTagger.tagMap.get(span)?.[OUTPUT_VALUE]) {
+          llmobs.annotate(span, { outputData: result })
+          if (llmobs.enabled) storage.enterWith(oldStore)
+        }
+
+        return result
+      } catch (e) {
+        if (llmobs.enabled) storage.enterWith(oldStore)
+        throw e
+      }
+    }
+
+    return this._tracer.wrap(name, spanOptions, wrapped)
+  }
+
+  annotate (span, options) {
+    if (!this.enabled) {
+      logger.warn(
+        'Annotate called while LLMObs is disabled. Not annotating span.'
+      )
+      return
+    }
+
+    if (!span) {
+      span = this.active()
+    }
+
+    if ((span && !options) && !(span instanceof Span)) {
+      options = span
+      span = this.active()
+    }
+
+    if (!span) {
+      logger.warn('No span provided and no active LLMObs-generated span found')
+      return
+    }
+    if (!options) {
+      logger.warn('No options provided for annotation.')
+      return
+    }
+
+    if (!LLMObsTagger.tagMap.has(span)) {
+      logger.warn('Span must be an LLMObs-generated span')
+      return
+    }
+    if (span._duration !== undefined) {
+      logger.warn('Cannot annotate a finished span')
+      return
+    }
+
+    const spanKind = LLMObsTagger.tagMap.get(span)[SPAN_KIND]
+    if (!spanKind) {
+      logger.warn('LLMObs span must have a span kind specified')
+      return
+    }
+
+    const { inputData, outputData, metadata, metrics, tags } = options
+
+    if (inputData || outputData) {
+      if (spanKind === 'llm') {
+        this._tagger.tagLLMIO(span, inputData, outputData)
+      } else if (spanKind === 'embedding') {
+        this._tagger.tagEmbeddingIO(span, inputData, outputData)
+      } else if (spanKind === 'retrieval') {
+        this._tagger.tagRetrievalIO(span, inputData, outputData)
+      } else {
+        this._tagger.tagTextIO(span, inputData, outputData)
+      }
+    }
+
+    if (metadata) {
+      this._tagger.tagMetadata(span, metadata)
+    }
+
+    if (metrics) {
+      this._tagger.tagMetrics(span, metrics)
+    }
+
+    if (tags) {
+      this._tagger.tagSpanTags(span, tags)
+    }
+  }
+
+  exportSpan (span) {
+    if (!this.enabled) {
+      logger.warn('Span exported while LLMObs is disabled. Span will not be exported.')
+      return
+    }
+
+    span = span || this.active()
+
+    if (!span) {
+      logger.warn('No span provided and no active LLMObs-generated span found')
+      return
+    }
+
+    if (!(span instanceof Span)) {
+      logger.warn('Span must be a valid Span object.')
+      return
+    }
+
+    if (!LLMObsTagger.tagMap.has(span)) {
+      logger.warn('Span must be an LLMObs-generated span')
+      return
+    }
+
+    try {
+      return {
+        traceId: span.context().toTraceId(true),
+        spanId: span.context().toSpanId()
+      }
+    } catch {
+      logger.warn('Faild to export span. Span must be a valid Span object.')
+      return undefined
```

Comment:
nit; just noticed we return `undefined` here and use only `return` above, i assume they do the same thing

Reply:
yep they do the same thing

Reply:
but i'll make it consistent